### DISCUSSION TITLE
Route tracing error on a Chinese car stereo

### DISCUSSION THREAD

**Discussion by thiagosouza2000** [osmandapp/OsmAnd#17402] [Maps Related]
Hello guys.

I'm trying to run OsmAnd on a Chinese head unit as the main offline navigator.

I've tried OsmAnd on these versions: nightly and full (downloaded the apk from osmand.net, as I don't want to get Google Play working with my Google account due to security concerns).

However, even with the proper offline maps downloaded, when I try to make a route for an address, I get a straight line (image bellow). If I get connected to the Internet, the routing works normally.

I tried to switch the location engine to Android (instead of Google Services). However in both options I can get my location, just not the route.

Is there anything I could do? 

Regards from Brazil.

![Screenshot_20230611-224946](https://github.com/osmandapp/OsmAnd/assets/28881448/50e7d13a-3f65-465b-abcc-468d5b54e73a)

**Comment by mikehgentry** [comment]
Maybe you can make brouter work?

https://osmand.net/docs/user/navigation/routing/brouter/

**Reply by thiagosouza2000** [reply]
> Maybe you can make brouter work?
> 
> https://osmand.net/docs/user/navigation/routing/brouter/

Thanks for replying. I gave brouter a try, but I after setting it up, I'm seeing a "brouter service is not available" error on OsmAnd.

Is there any other offline solution? 

If not, maybe I should consider get help with this new error and persist on brouter.

**Reply by mikehgentry** [reply]
I've seen that occasionally on my devices (your average phones...)

I take it you've installed and run brouter, and downloaded the appropriate routing graphs. If so I'd maybe see if excluding brouter from being closed in the background (if it lets you), rebooting, and running it again helps.

Other than that no idea I'm afraid.

**Comment by thiagosouza2000** [comment]
Good news! After turning off "battery optimization" (actually works as a RAM free space control) for Brouter and rebooting the device, the routing is working. Thanks!
